core-bundle/contao/library/Contao/Controller.php

```diff
@@ -514,13 +515,27 @@ public static function getArticle($varId, $blnMultiMode=false, $blnIsInsertTag=f
 	/**
 	 * Generate a content element and return it as string
 	 *
-	 * @param mixed  $intId     A content element ID or a Model object
+	 * @param mixed  $intId     A content element ID, a Model object or a ContentElementReference
 	 * @param string $strColumn The column the element is in
 	 *
 	 * @return string The content element HTML markup
 	 */
 	public static function getContentElement($intId, $strColumn='main')
 	{
+		$contentElementReference = null;
+
+		if ($intId instanceof ContentElementReference)
+		{
+			if (\func_num_args() > 1)
+			{
+				throw new \InvalidArgumentException('Passing a column name is not supported when using a ContentElementReference.');
+			}
+
+			$contentElementReference = $intId;
+			$strColumn = $contentElementReference->getSection();
+			$intId = $contentElementReference->getContentModel();
```

Comment:
```suggestion
			$strColumn = $contentElementReference->attributes['section'];
			$intId = $contentElementReference->attributes['contentModel'];
```

`ControllerReference::$attributes` is a public property, so we could use it directly and remove the getters in out `ContentElementReference` class again. WDYT?